docs: Update README (#9)
unitaryfoundation/jeff: README.md

<p align="center">
  <img src="https://raw.githubusercontent.com/unitaryfoundation/jeff/docs/images/JEFF_github_logo.png#gh-light-mode-only" width="700px">
    <!--
    Use a relative import for the dark mode image. When loading on alternative services (such as PyPI), this
    will fail automatically and show nothing.
    -->
    <img src="./docs/images/JEFF_github_logo_inverted.png#gh-dark-mode-only" width="700px" onerror="this.style.display='none'" alt=""/>
</p>

*Jeff* is a structured, extensible, and simple interchange format for quantum compilers. Designed
to foster collaboration and accelerate research by providing a common language
for compilers to communicate.

## Features

* Structured representation: First class support for hybrid, structured, and dynamic quantum
  programs. Share the logical intent of a program, not just a low-level gate sequence.

> * Efficient: *Jeff* is built to be a high-performance in-memory representation, making it

Author: ```suggestion
* Efficient: *Jeff* is built to be a high-performant, compact, in-memory representation, making it
```

>   fast and easy for compilers to parse, transform, and export.

Author: ```suggestion
  fast and easy for compilers to load, transform, and export.
```

* Extensible by design: The *Jeff* schema is designed to allow maximum extensibility --- it can be
  easily extended with new features or metadata, ensuring it can grow with the field.

* Bring your own compiler framework: *Jeff* provides a neutral ground for different quantum
  compilers (such as Catalyst, TKET, UCC, and others) to share passes, interoperate seamlessly
  on a single workflow.

## Motivation

The quantum software field is at a critical juncture --- as quantum algorithms become increasingly
sophisticated, quantum program representation is evolving from low-level, gate-centric approaches
to more complex representations that incorporate classical processing, structure, and dynamism.

In particular,

- There are some algorithmic patterns we cannot represent without structure and dynamism

>   (for example, repeat-until-success).

Author: ```suggestion
  (for example, repeat-until-success and quantum error correction).
```

- Algorithms inherently contain structure --- by preserving this structure during optimization,
  we can massively improve scalability.

- Thinking beyond simple circuits will allow us to do new things and progress the field.

Compilation and optimization of such programs is crucial to support quantum hardware execution,
however current de-facto interchange formats (such as QIR, and OpenQASM) either lacks the explicit
structure needed for advanced, high-level optimizations, or was not designed for program
optimization.

***This creates a bottleneck for collaboration and makes it difficult to develop and share
sophisticated compiler passes.***

Unlike other common quantum program formats, *Jeff* is not a human-readable source language or a

> low-level intermediate representation. It's an in-memory representation designed for

Reviewer: For folks less in the compiler space, they might view this as a negative without understanding the performance impacts (or even feasibility) of sticking with human-readable formats. This idea is somewhat implicit in the points above, but might be worth making explicit if that is a true design principle?

Just a suggestion/not at all blocking

Author: @bachase on @ss2165's suggestion, I'm going to add another doc for FAQs, and include this there

<p align="center">
  <img src="https://raw.githubusercontent.com/unitaryfoundation/jeff/docs/images/JEFF_github_logo.png#gh-light-mode-only" width="700px">
    <!--
    Use a relative import for the dark mode image. When loading on alternative services (such as PyPI), this
    will fail automatically and show nothing.
    -->
    <img src="./docs/images/JEFF_github_logo_inverted.png#gh-dark-mode-only" width="700px" onerror="this.style.display='none'" alt=""/>

> </p>

Author: Note that this will not render correctly until the images are available on `main`
